File: frontend/claims-app/src/app/features/claims/models/claim.ts
import { ClaimStatus } from './claim-status';

export interface Claim {
  id: number;
  policyNumber: string;
  insuredName: string;
  claimType: string;
  estimatedAmount: number;
  status: ClaimStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateClaimRequest {
  policyNumber: string;
  insuredName: string;
  claimType: string;
  estimatedAmount: number;
}

export interface UpdateClaimRequest {
  policyNumber: string;
  insuredName: string;
  claimType: string;
  estimatedAmount: number;
}

export interface ChangeClaimStatusRequest {
  newStatus: ClaimStatus;
}

export interface ClaimStatusHistory {
  id: number;
  claimId: number;
  previousStatus: ClaimStatus | null;
  newStatus: ClaimStatus;
  changedAt: string;
}

File: src/Claims.Domain/Entities/Claim.cs
using Claims.Domain.Enums;

namespace Claims.Domain.Entities;

public class Claim
{
    public int Id { get; set; }

    public string PolicyNumber { get; set; } = string.Empty;

    public string InsuredName { get; set; } = string.Empty;

    public string ClaimType { get; set; } = string.Empty;

    public decimal EstimatedAmount { get; set; }

    public ClaimStatus Status { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public ICollection<ClaimStatusHistory> StatusHistory { get; set; } = new List<ClaimStatusHistory>();
}